### PR TITLE
Fix various errors on flask application

### PR DESCRIPTION
**Pull request by mmphego**
@SudKul these are some of the errors picked up on the repo, which as a mentor has surfaced several times on [knowledge](https://knowledge.udacity.com/?nanodegree=nd064-beta&page=1&project=770&rubric=2909)

### CODE REVIEW

**Comment by SudKul** [comment]
Thanks @mmphego, for taking the initiative to keep the repo up to date. 
I looked into the files edited in this PR. Can you share your points on:

1. Why some imports have been removed from *Project_Starter_Files-Building_a_Metrics_Dashboard/reference-app/trial/app.py* file?
2. Should the starter code provide the `def trace()` function definition in the *Project_Starter_Files-Building_a_Metrics_Dashboard/reference-app/trial/app.py* file?
3. Why some packages have been removed from *Project_Starter_Files-Building_a_Metrics_Dashboard/reference-app/trial/requirements.txt*?

The remaining edits look good to me.

File: Project_Starter_Files-Building_a_Metrics_Dashboard/reference-app/trial/app.py [hunk 1]
```diff
@@ -1,73 +1,93 @@
...
 
+
+from flask import Flask, jsonify, render_template
+from flask_opentracing import FlaskTracing
 from jaeger_client import Config
 from jaeger_client.metrics.prometheus import PrometheusMetricsFactory
-from opentelemetry import trace
-from opentelemetry.exporter import jaeger
-from opentelemetry.sdk.trace import TracerProvider
-from opentelemetry.sdk.trace.export import BatchExportSpanProcessor
```
Comment (mmphego) [inline Project_Starter_Files-Building_a_Metrics_Dashboard/reference-app/trial/app.py:8]: > Why some imports have been removed from Project_Starter_Files-Building_a_Metrics_Dashboard/reference-app/trial/app.py file?

@SudKul, 
- These imports aren't well explained as to why they are needed!
- They add complexity to the project and confusions for students as they aren't justified on the lessons
- Replaced all imports with `PrometheusMetrics` which does not or less the same

File: Project_Starter_Files-Building_a_Metrics_Dashboard/reference-app/trial/app.py [hunk 2]
```diff
@@ -1,73 +1,93 @@
...
 def homepage():
     return render_template("main.html")
-    with tracer.start_span('get-python-jobs') as span:
-        homepages = []
-        res = requests.get('https://jobs.github.com/positions.json?description=python')
-        span.set_tag('first-tag', len(res.json()))
+
+
+@app.route("/trace")
+def trace():
```
Comment (mmphego) [inline Project_Starter_Files-Building_a_Metrics_Dashboard/reference-app/trial/app.py:55]: @SudKul this function is decoupled from `homepage` function defined above. Also note that in the `homepage` there's a syntax error; you cannot have logic defined after `return` is called

File: Project_Starter_Files-Building_a_Metrics_Dashboard/reference-app/trial/requirements.txt [hunk 3]
```diff
@@ -1,11 +1,12 @@
...
 jaeger-client==4.3.0
 opentelemetry-api==0.12b0
-opentelemetry-sdk==0.12b0
-opentelemetry-instrumentation==0.12b0
+opentelemetry-exporter-jaeger==0.12b0
 opentelemetry-instrumentation-flask==0.12b0
 opentelemetry-instrumentation-requests==0.12b0
-opentelemetry-exporter-jaeger==0.12b0
+opentelemetry-instrumentation==0.12b0
+opentelemetry-sdk==0.12b0
```
Comment (mmphego) [inline Project_Starter_Files-Building_a_Metrics_Dashboard/reference-app/trial/requirements.txt:9]: @SudKul No imports where removed, just sorted by alpha

**Review by mmphego — COMMENTED** [review]
If you aren't happy with the removal of the imports, I can re-add them...Let me know

File: Project_Starter_Files-Building_a_Metrics_Dashboard/reference-app/trial/requirements.txt (hunk 3 above)
Comment (SudKul) [reply Project_Starter_Files-Building_a_Metrics_Dashboard/reference-app/trial/requirements.txt:9]: Perfect! Thanks

File: Project_Starter_Files-Building_a_Metrics_Dashboard/reference-app/trial/app.py (hunk 2 above)
Comment (SudKul) [reply Project_Starter_Files-Building_a_Metrics_Dashboard/reference-app/trial/app.py:55]: Yeah, you are right. The function execution will not proceed past the `return` statement.

**Comment by SudKul** [comment]
Thanks, @mmphego 
I am merging your edits.

**Pull request merged**